native/src/utils/DefaultDataContainer.ts

```diff
@@ -179,22 +141,12 @@ class DefaultDataContainer implements DataContainer {
     resourceCache: LanguageResourceCacheStateType,
   ): Promise<void> => {
     const context = new DatabaseContext(city)
-    const cache: Cache<CityResourceCacheStateType> = this.caches.resourceCache
-    const previousResourceCache = cache.getCached(context)
-
-    if (!previousResourceCache) {
-      await cache.cache(
```

Comment:
This was the culprit @bahaaTuffaha of the cache not correctly updating on first load. We were setting the resource cache on the wrong cache instance since this is a local copy and not `this.caches.resourceCache` thats later read again.